lib/apartment/adapters/sqlite3_adapter.rb

```diff
@@ -13,6 +13,9 @@ def initialize(config)
         @default_dir = File.expand_path(File.dirname(config[:database]))
 
         super
+
+        Apartment.connection_handler = ActiveRecord::ConnectionAdapters::ConnectionHandler.new
+        reset
```

Comment:
when I'm contributing to open source I try to make small & concise PRs that address a single problem, as it makes it easier for maintainers to review.

then address any feedback from maintainers w.r.t. refactoring / style latterly.